tests/test_dtype/test_npy/test_float.py

```diff
@@ -12,6 +12,15 @@ def scalar_equals(self, scalar1: object, scalar2: object) -> bool:
             return True
         return super().scalar_equals(scalar1, scalar2)
 
+    hex_nan_params: tuple[str, ...] = ()
+
+    def test_hex_nan(self, hex_nan_params: str) -> None:
```

Comment:
This should probably also test a non-nan value:

`0x4048f5c3` for `3.14`

Reply:
I test for some non-nans in https://github.com/zarr-developers/zarr-python/pull/2874/commits/8c90d2ca827de0846f9ce65e045b24e5b5682527